### DISCUSSION TITLE
The openid paperless redirect_uris is not correct in the documentation.

### DISCUSSION THREAD

**Discussion by saket1999** [authelia/authelia#7468] [Documentation Feedback] [answered]
In the documentation, the redirect_uris of paperless is `https://paperless.example.com/accounts/authelia/login/callback` but it was not working, instead this is the correct uri `https://paperless.example.com/accounts/oidc/authelia/login/callback/`.

I was getting this error while using the uri from the docs

````
authelia        | time="2024-06-30T14:46:14+05:30" level=error msg="Authorization Request failed with error: The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered 'redirect_uris'. The 'redirect_uris' registered with OAuth 2.0 Client with id 'paperless' match 'redirect_uri' value 'https://paperless.example.com/accounts/oidc/authelia/login/callback/'." method=GET path=/api/oidc/authorization remote_ip="2401:**************:9e10" stack="github.com/authelia/authelia/v4/internal/handlers/handler_oidc_authorization.go:41           OpenIDConnectAuthorization\ngithub.com/authelia/authelia/v4/internal/middlewares/http_to_authelia_handler_adaptor.go:114 handleRouter.NewHTTPToAutheliaHandlerAdaptor.func21\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54                            handleRouter.(*BridgeBuilder).Build.func16.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:91                           SecurityHeadersNoStore.func1\ngithub.com/valyala/fasthttp@v1.52.0/userdata.go:57
         (*userData).Get\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:30                           SecurityHeadersBase.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/cors.go:216                             handleRouter.(*CORSPolicy).Middleware.func22\ngithub.com/fasthttp/router@v1.5.0/router.go:441                                              (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14                       handleRouter.LogRequest.func40\ngithub.com/authelia/authelia/v4/internal/middlewares/errors.go:38                            RecoverPanic.func1\ngithub.com/valyala/fasthttp@v1.52.0/server.go:2374                                           (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:224                                        (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:196
                         (*workerPool).getCh.func1\nruntime/asm_amd64.s:1695                                                                     goexit"
````

I have changed my domain to example.com and masked the ip.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It appears that the redirect URI specified in the documentation for Paperless is incorrect. The correct redirect URI should be `https://paperless.example.com/accounts/oidc/authelia/login/callback/` instead of `https://paperless.example.com/accounts/authelia/login/callback`.

The error message you received indicates that the `redirect_uri` parameter does not match any of the pre-registered `redirect_uris` for the OAuth 2.0 Client with id 'paperless'. This mismatch is causing the authorization request to fail.

To resolve this issue, you should update the `redirect_uris` in your Authelia configuration to include the correct URI:

```yaml
clients:
  - client_id: 'paperless'
    client_name: 'Paperless'
    client_secret: 'your_client_secret'
    redirect_uris:
      - 'https://paperless.example.com/accounts/oidc/authelia/login/callback/'
    scopes:
      - 'openid'
      - 'profile'
      - 'email'
    userinfo_signed_response_alg: 'none'
    token_endpoint_auth_method: 'client_secret_post'
```

Make sure to replace `'your_client_secret'` with your actual client secret.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by ados8** [comment]
This doesn't work for me, I have setup other OpenID apps fine.
No matter the settings I get redirect URI Error.

**Reply by james-d-elliott** [reply]
The error includes the redirect URI requested by the client (paperless). You just need to copy it.

**Comment by james-d-elliott** [comment, marked as answer]
Resolved in the docs
